dev/Repeater/APITests/LayoutTests.cs

```diff
@@ -192,6 +192,33 @@ public void ValidateNonVirtualLayoutDoesNotGetMeasuredForViewportChanges()
             });
         }
 
+        [TestMethod]
+        public void ValidateStackLayoutDoesNotRetainIncorrectMinorWidth()
+        {
+            RunOnUIThread.Execute(() =>
+            {
+                var repeater = new ItemsRepeater();
+                var stackLayout = new StackLayout();
+                repeater.Layout = stackLayout;
+                repeater.ItemsSource = Enumerable.Range(0, 1);
+                repeater.ItemTemplate = (DataTemplate)XamlReader.Load(
+                    @"<DataTemplate  xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'>
+                         <Button Content='{Binding}' Height='100' />
+                    </DataTemplate>");
+                var scrollViewer = new ScrollViewer() {
+                    Content = repeater
+                };
+                scrollViewer.Width = 300;
+                Content = scrollViewer;
+                Content.UpdateLayout();
+
+                repeater.Measure(new Size(600, 100));
```

Comment:
I think you need to perform multiple measure's before calling UpdateLayout to reproduce the issue.

Reply:
UpdateLayout results in another Measure call (with the proper sizing). When running this test without the fix, the repeater was being sized to 600 erroneously as a result.

Reply:
UpdateLayout will run measure/arrange - It might be best to avoid calling UpdateLayout and instead just call measure/arrange directly on the element. DesiredSize should reflect changes after Measure and ActualWidth should get updated after Arrange call.

Reply:
Manually calling Measure and Arrange for some reason does not update the ActualWidth property. Only by calling UpdateLayout I managed to get the ActualWidth property updated.